### DISCUSSION TITLE
WebRTC subscription load/test tool

### DISCUSSION THREAD

**Discussion by benbullnz** [bluenviron/mediamtx#2305] [Questions]
Hi all, 

Would someone be able to help me with a basic standalone go program which will subscribe to a mediamtx webrtc stream with whep? 

I originally had this all working with the first webrtc release using web sockets, but it appears that the approach was altered a few versions later and my adaption now no longer works with the latest code.

The authors have again done exceptional work in adding these features and there is function called TestWebRTCRead() in webrtc_manager_test.go available. However after a few attempts at removing the "test" features, I am still struggling to migrate this into its own standalone subscripton test tool. The underlying track channel need not go anywhere, I effectively need it as a subscription load test tool. Any help would be appreciated.

**Comment by aler9** [comment]
Hello, a Golang-based, standalone WebRTC client, compliant with WHIP and WHEP, is now available in folder [internal/protocols/webrtc](https://github.com/bluenviron/mediamtx/blob/main/internal/protocols/webrtc/whip_client.go).

**Reply by benbullnz** [reply]
@aler9 I'd be really appreciateive if you could you provide a basic example which can be used to subscribe to a whep stream and push it to another with whip using this standalone client

[a comment by an automated account is not shown]
